src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -32,12 +32,85 @@ public Tag(String tagName, TagStatus tagStatus) {
         checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
         this.tagName = tagName;
         this.tagStatus = tagStatus;
+        this.tagType = getTagType(tagStatus);
+    }
+
+    /**
+     * Creates a new Tag with specified tagName and tagStatus.
+     *
+     * @param tagName Name of the tag to be created.
+     * @param tagStatus Status of the tag.
+     * @return A new tag of specific type corresponding to the TagStatus input.
+     */
+    public static Tag createTag(String tagName, TagStatus tagStatus) {
+        requireNonNull(tagName);
+        // require the tagStatus not to be null for now
+        // in the future, a null tagStatus input should be set to INCOMPLETE_GOOD
+        // by default
+        requireNonNull(tagStatus);
+        checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
+        TagType tagType = getTagType(tagStatus);
+
+        switch (tagType) {
+        case TUTORIAL:
+            return new TutorialTag(tagName, tagStatus);
+        case ATTENDANCE:
+            return new AttendanceTag(tagName, tagStatus);
+        case ASSIGNMENT:
```

Comment:
Should you move return new AssignmentTag(tagName, tagStatus); above to case ASSIGNMENT, and let default return null instead?

Reply:
sure sure that would work better!